bin/setup

```diff
@@ -0,0 +1,127 @@
+#!/usr/bin/env bash
+
+##
+# Setup script to symlink configuration and dot files from this repository
+# to proper places in the system.
+#
+# Script is idempotent. It is same to run it several times.
+#
+# TODO:
+
+# Stop immediately if any external program returns a non-zero exit status.
+# Short: set -e
+set -o errexit
+# Treat unset variables as an error when substituting. Short: set -u
+set -o nounset
+# If set, the return value of a pipeline is the value of the last (rightmost)
+# command to exit with a non-zero status, or zero if all commands in the
+# pipeline exit successfully. This option is disabled by default.
+set -o pipefail
+
+# Uncomment for debugging
+#
+# Disable file name generation using metacharacters (globbing).
+# Short: set -f
+#set -o noglob
+# Prints shell input lines as they are read.
+# Short: set -v
+#set -o verbose
+# Print command traces before executing command.
+# Short: set -x
+#set -o xtrace
+
+
+readonly __DIR__="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+readonly __LIB__="${__DIR__}/../lib-bash"
+
+readonly __CONFIG__="${__DIR__}/../CONFIG"
+
+
+readonly USER_HOME_DIR_PLACEHOLDER="USER_PLACEHOLDER"
+
+readonly DEFAULT_USER="$USER"
+declare  USER_NAME="${DEFAULT_USER}"
+
+readonly DEFAULT_HOME_DIR_PATH="$HOME"
+declare  HOME_DIR_PATH="${DEFAULT_HOME_DIR_PATH}"
+
+readonly DEFAULT_CONFIG_DIR_NAME=".config"
+declare  CONFIG_DIR_NAME="${DEFAULT_CONFIG_DIR_NAME}"
+
+
+main() {
+  while getopts "H:U:h" _FLAG; do
+    case $_FLAG in
+      U)
+        USER_NAME="$OPTARG"
+        ;;
+      H)
+        HOME_DIR_PATH="$OPTARG"
+        ;;
+      h)
+        usage
+        exit 0
+        ;;
+      *)
+        usage
+        exit 1
+    esac
+  done
+
+
+  do_symlinking
+
+  exit 0
+}
+
+usage() {
+  cat <<-EOS
+
+  Setup script to symlink configuration and dot files from this repository
+  to proper places in the system.
```

Comment:
This partially repeats what is written in the comment.
If you want to look very cool :sunglasses: , you can build complete help message from the documentation.
See example here: https://github.com/greyblake/beep-alarm/blob/master/bin/beep-alarm#L175

Reply:
It is very cool trick! But it seems a bit hacky! I leave my approach as yours wouldn't work for me because I need precise control over how my documentation looks like and I want to interpolate variables in it which means it can not go into comments at the header of the script.